pkg/consensus/client.go

```diff
@@ -94,27 +103,39 @@ func (c *Client) loadCurrentContext(ctx context.Context, currentSlot types.Slot,
 	return nil
 }
 
+func (c *Client) GetProposerCache(ctx context.Context, slot types.Slot) (ValidatorInfo, bool) {
```

Comment:
this function does not get the "proposer cache" and instead provides the `ValidatorInfo` for a given proposer
```suggestion
func (c *Client) GetProposer(ctx context.Context, slot types.Slot) (ValidatorInfo, bool) {
```